connector/connect/common/src/main/protobuf/spark/connect/base.proto

```diff
@@ -183,6 +183,141 @@ message ExecutePlanResponse {
   }
 }
 
+// The placeholder for the config request and response when the values can be optional.
+message OptionalValue {
+  optional string value = 1;
+}
+
+// The key-value pair for the config request and response.
+message KeyValue {
+  string key = 1;
+  string value = 2;
+}
+
+// The key-value pair for the config request and response when the value can be optional.
+message OptionalKeyValue {
+  string key = 1;
+  OptionalValue value = 2;
+}
```

Comment:
Nit that please follow the proto style guide to mark each of the field by `(Required)` or `(Optional)`.